src/sign-typed-data.ts

```diff
@@ -174,7 +174,7 @@ function encodeField(
     if (typeof value === 'number') {
       value = numberToBuffer(value);
     } else if (isHexString(value)) {
-      value = numberToBuffer(parseInt(value, 16));
```

Comment:
I believe (based on the comment in the [PR that changed this](https://github.com/MetaMask/eth-sig-util/commit/9f01c9d7922b717ddda3aa894c38fbba623e8bdf#diff-39b2554fd18da165b59a6351b1aafff3714e2a80c1435f2de9706355b4d32351R128)) the change from `Buffer.from` to `numberToBuffer` was intentional, but I'm not sure why. @adonesky1 Would you be willing to elaborate on this?

Reply:
@mcmire this change was intentional. And apparently our testing was not sufficient to expose the regression here. This change was made in response to feedback provided [here](https://github.com/MetaMask/eth-sig-util/pull/260#pullrequestreview-1062614981) to a much larger set of changes in this suite of related functions. This change does seem to get us back on track. I am a bit concerned now however that there may be other regressions from [this PR ](https://github.com/MetaMask/eth-sig-util/pull/260)still lurking, since our test suite didn't catch this one...

Reply:
cc @kumavis